src/NuGet.Indexing/DownloadsByVersion.cs

```diff
@@ -48,8 +41,13 @@ public int Total
 
             set
             {
+                if (_downloadsByVersion.ContainsKey(version))
+                {
+                    _total = _total - _downloadsByVersion[version];
```

Comment:
If the issue was caused by concurrency why not make this read/writes thread safe?